control-plane/pkg/core/config/utils.go

```diff
@@ -97,7 +97,7 @@ func EventPoliciesFromAppliedEventPoliciesStatus(status duck.AppliedEventPolicie
 		eventPolicies = append(eventPolicies, contractPolicy)
 	}
 
-	if len(eventPolicies) == 0 {
+	if len(eventPolicies) == 0 && features.IsOIDCAuthentication() {
```

Comment:
it seems that this condition needs to be true from the very beginning and return early ?

Reply:
Updated it (anyhow, as we added only Ready EventPolicies and an EP only is Ready when OIDC is enabled this should result in the same - but this safes some cycles)